### PR TITLE
ci: update circle-ci workflows

### PR DESCRIPTION
**Pull request by sibiraj-s**
## I'm submitting a...

```
[ ] Bug Fix
[ ] Feature
[x] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

Update Circle CI Workflows.
This separates deployment as a separate workflow and runs only on master branch with tagged commits.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,8 +44,16 @@ jobs:
...
+
+  deploy:
+    <<: *defaults
+    steps:
+      - attach_workspace: *attach_options
+      - run:
+          name: Deploying package to npm registry
           command: |
             # CIRCLE_TAG indicates whether the job runs for the tag (as opposed to: the commit on the branch)
             if [ -z "${CIRCLE_TAG}" ]; then
```
Comment (Sibiraj-S) [inline .circleci/config.yml]: Can this be removed. In current workflow, deployment happens only in master with tags matching `/v.*/`.

Comment (Sibiraj-S) [reply .circleci/config.yml]:
```

Comment (Sibiraj-S) [reply .circleci/config.yml]: The deployment already is configured for tagged builds.

      tags:
          only: /v.*/

 I think checking that again is not necessary.

**Comment by sibiraj-s** [comment]
@dherges

**Comment by dherges** [comment]
No, this broke the whole publisjhing workflow ... 

```#!/bin/bash -eo pipefail
# CIRCLE_TAG indicates whether the job runs for the tag (as opposed to: the commit on the branch)
if [ -z "${CIRCLE_TAG}" ]; then
  echo "Skipping deployment. This build job runs for a branch."
  exit 0
fi
echo "Deploying..."
# The git tag that Circle CI checked out
GIT_TAG=$(git tag -l --points-at HEAD)
# The npm tag for publishing this release either as "latest" or "next"
NPM_TAG=latest
if [[ "${GIT_TAG}" =~ v([0-9]+)\.([0-9]+)\.([0-9]+)-([a-zA-Z]+)\..* ]]; then
  NPM_TAG=next
fi
# Build, pack and publish to npm registry
echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
yarn publish:ci
pushd dist
npm publish --tag ${NPM_TAG}
Skipping deployment. This build job runs for a branch.
```

jobs for the tags are no longer being run. I am investigating...

**Comment by dherges** [comment]
Found it.

> Executing Workflows for a Git Tag
> CircleCI does not run workflows for tags unless you explicitly specify tag filters. Additionally, if a job requires any other jobs (directly or indirectly), you must use regular expressions to specify tag filters for those jobs.

https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag

which means we have to add the "filters" section to each job in the [`.config.yml`](https://github.com/ng-packagr/ng-packagr/blob/master/.circleci/config.yml#L83-L102)

**Comment by sibiraj-s** [comment]
Oops. Sorry @dherges . didn't notice that. I didn't alter much. there are few scenarios where deployment was done from branches other than master, hence I moved it to workflows restricting the deployment only to master.

**Pull request merged**